src/integration-test/integration/ODataClient.test.ts

```diff
@@ -0,0 +1,112 @@
+
+  let reportId: string;
+  let oDataItem: ODataItem;
+  let mappingId: string;
+
+  before(async () => {
+    const newMapping: MappingCreate = {
+      mappingName: "Test",
+    };
+    const mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    expect(mapping.mappingName).to.be.eq("Test");
+    mappingId = mapping.id;
+
+    const newGroup: GroupCreate = {
+      groupName: "Test",
+      query: "select * from biscore.element limit 10",
+    };
+    const group = await mappingsClient.createGroup(accessToken, testIModel.id, mapping.id, newGroup);
+    expect(group).to.not.be.undefined;
+    expect(group.groupName).to.be.eq("Test");
+
+    const newReport: ReportCreate = {
+      displayName: "Test",
+      projectId,
+    };
+    const report = await reportsClient.createReport(accessToken, newReport);
+    expect(report).to.not.be.undefined;
+    expect(report.displayName).to.be.eq("Test");
+    reportId = report.id;
+
+    const newReportMapping: ReportMappingCreate = {
+      mappingId: mapping.id,
+      imodelId: testIModel.id,
+    };
+    const reportMapping = await reportsClient.createReportMapping(accessToken, report.id, newReportMapping);
+    expect(reportMapping).to.not.be.undefined;
+    expect(reportMapping.mappingId).to.be.eq(mapping.id);
+
+    const extraction = await extractionClient.runExtraction(accessToken, testIModel.id);
+    expect(extraction).to.not.be.undefined;
+
+    let state = ExtractorState.Queued;
+    let status: ExtractionStatus;
+    for (const start = performance.now(); performance.now() - start < 6 * 60 * 1000; await sleep(3000)) {
+      status = await extractionClient.getExtractionStatus(accessToken, extraction.id);
+      state = status.state;
+      if(state !== ExtractorState.Queued && state.valueOf() !== ExtractorState.Running)
+        break;
+    }
+    expect(state).to.be.eq(ExtractorState.Succeeded);
+
+    const oDataResponse = await oDataClient.getODataReport(accessToken, reportId);
+    expect(oDataResponse).to.not.be.undefined;
+    expect(oDataResponse.value).to.not.be.empty;
+    oDataItem = oDataResponse.value[0];
+
+  });
+
+  after(async () => {
+    await mappingsClient.deleteMapping(accessToken, testIModel.id, mappingId);
+    await reportsClient.deleteReport(accessToken, reportId);
+
+    await testIModelGroup.cleanupIModels();
+  });
+
+  it("get OData report", async () => {
+    const oDataResponse = await oDataClient.getODataReport(accessToken, reportId);
+    expect(oDataResponse).to.not.be.undefined;
+    expect(oDataResponse["@odata.context"]).to.not.be.empty;
+  });
+
+  it("get OData report metadata", async () => {
+    const oDataResponse = await oDataClient.getODataReportMetadata(accessToken, reportId);
+    expect(oDataResponse).to.not.be.undefined;
+    expect(oDataResponse).to.not.be.empty;
+    expect(oDataResponse[0].name).to.not.be.empty;
```

Comment:
You can add assertions on properties. `ECInstanceId`, `ECClassId`, `UserLabel`, `BBowLow`, and `BBoxHigh` should exist. All of their types should be set to `Edm.String`.

Reply:
asserting all values of oDataResponse